pydantic_ai_slim/pydantic_ai/models/ollama.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations as _annotations
+
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass, field
+
+from httpx import AsyncClient as AsyncHTTPClient
+
+from . import (
+    AbstractToolDefinition,
+    AgentModel,
+    Model,
+    cached_async_http_client,
+    check_allow_model_requests,
+)
+
+try:
+    from openai import AsyncOpenAI
+    from openai.types import chat
+except ImportError as e:
+    raise ImportError(
+        'Please install `openai` to use the OpenAI model, '
+        "you can use the `openai` optional group — `pip install 'pydantic-ai[openai]'`"
+    ) from e
+
+# Only required in the below typing is enforced
+# try:
+#     from ollama import list, ListResponse
+# except ImportError as e:
+#     raise ImportError(
+#         'Please install `ollama` to use the Ollama model, '
+#         "you can use the `ollama` optional group — `pip install 'pydantic-ai[ollama]'`"
+#     ) from e
+
+
+from .openai import OpenAIAgentModel
+
+# NB: Currently commented out but this is
+# a possible approach to type the ollama model names
+# # This is really nasty, but ollama doesn't
+# # expose a type of all available models, and as
+# # there are 100's and being udpated all the time,
+# # it seems difficult to maintain a list of all possible
+# # versions statically.
+# # This code gets the ones that are _currently_ available
+# # to the user as they are ones that they have already downloaded
+# ollama_ls: ListResponse = list()
+# ollama_available_models: list[str] = [x.model for x in ollama_ls.models]
+# OllamaModelName = Literal[tuple(ollama_available_models)]
```

Comment:
My suggestion is we type `OllamaModelName` as

```py
CommonOllamaModelNames = Literal['llama3.3', 'llama3.2', 'qwq', ...]
"""This contains just the most common ollama models.

For a full list see [ollama.com/library](https://ollama.com/library).
"""
OllamaModelName = Union[CommonOllamaModelNames, str]
"""Possible ollama models.

Since Ollama supports hundreds of models, we explicitly list the most models but
allow any name in the type hints.
"""
```

Then add the most widely used 8 - 15 models to `CommonOllamaModelNames`.

I checked and with both pycharm and vscode this provides auto-complete while still allowing any string:

<img width="758" alt="image" src="https://github.com/user-attachments/assets/9849bae5-338c-4503-baec-7ba60b7e9f78">

<img width="671" alt="image" src="https://github.com/user-attachments/assets/eb580516-042b-487b-bcff-74b0b762decb">